samples/csharp/MicrosoftEduGraphSamples/Utilities/GlobalMethods.cs

```diff
@@ -41,5 +52,31 @@ public static void ValidateConfiguration(IConfiguration configuration)
             }
         }
 
+        /// <summary>
+        /// Workflow to Publish Assignments
+        /// </summary>
+        /// <param name="assignmentId">Assignment id</param>
+        public static async Task PublishAssignmentsAsync(Microsoft.Graph.Beta.GraphServiceClient graphClient, string assignmentId)
+        {
+            int retries = 0;
+
+            // Teacher publishes the assignment to make it appears in the student's list
+            var assignment = await MicrosoftGraphSDK.Assignment.PublishAsync(graphClient, _config["classId"], assignmentId);
+            Console.WriteLine($"Assignment {assignment.Id} publish in process");
+
+            // Verify assignment state, publish is completed until state equals "Assigned"
+            while (assignment.Status != EducationAssignmentStatus.Assigned && retries <= MAX_RETRIES)
+            {
+                // Print . in the log to show that the call is being retried
+                Console.WriteLine(".");
+
+                assignment = await MicrosoftGraphSDK.Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+
+                // If you are calling this code pattern in Backend agent of your service, then you want to retry the work after some time. The sleep here is just an example to emulate the delay
```

Comment:
illustrate comment further by naming some of the popular retry policies out there in use like linear, step function, exponential.

Reply:
added